tpi/templates/main.tf

```
@@ -23,4 +23,8 @@ resource "iterative_machine" "{{name}}" {
 {{ startup_script }}
   EOF
   {% endif %}
+  {% if metadata %}metadata = {
+    {% for key, value in metadata.items() %}{{key}} = "{{value}}"
+    {% endfor %}
+  }{% endif %}
```

i think if you wanna suppress newlines this should work...

```suggestion
  {% endif %}{% if metadata %}
  metadata = {{% for key, value in metadata.items() %}
    {{key}} = "{{value}}"{% endfor %}
  }{% endif %}
```

---

Yes, it works perfectly, but `metadata = {{% for` is a syntax error because it interprets the two curly braces as the beginning of a variable, while the first should be literal and the second should be part of the beginning of a block.

---

well there's a bunch of whitespace elsewhere anyway lol.